sdk/src/generated_sdk.rs

```diff
@@ -390,6 +468,136 @@ pub mod types {
+    /// `RangeFrom` types in the standard library. Those cover `(..end)`,
+    /// `(start..end)`, and `(start..)` respectively.
+    #[derive(Clone, Debug, Deserialize, Serialize, schemars :: JsonSchema)]
+    #[serde(tag = "type")]
+    pub enum BinRangeint8 {
+        /// A range unbounded below and exclusively above, `..end`.
+        #[serde(rename = "range_to")]
+        RangeTo { end: i8 },
+        /// A range bounded inclusively below and exclusively above,
+        /// `start..end`.
+        #[serde(rename = "range")]
+        Range { end: i8, start: i8 },
+        /// A range bounded inclusively below and unbounded above, `start..`.
+        #[serde(rename = "range_from")]
+        RangeFrom { start: i8 },
+    }
+
+    impl From<&BinRangeint8> for BinRangeint8 {
+        fn from(value: &BinRangeint8) -> Self {
+            value.clone()
+        }
+    }
+
+    /// A type storing a range over `T`.
+    ///
+    /// This type supports ranges similar to the `RangeTo`, `Range` and
+    /// `RangeFrom` types in the standard library. Those cover `(..end)`,
+    /// `(start..end)`, and `(start..)` respectively.
+    #[derive(Clone, Debug, Deserialize, Serialize, schemars :: JsonSchema)]
+    #[serde(tag = "type")]
+    pub enum BinRangeuint16 {
+        /// A range unbounded below and exclusively above, `..end`.
+        #[serde(rename = "range_to")]
+        RangeTo { end: u16 },
+        /// A range bounded inclusively below and exclusively above,
+        /// `start..end`.
+        #[serde(rename = "range")]
+        Range { end: u16, start: u16 },
+        /// A range bounded inclusively below and unbounded above, `start..`.
+        #[serde(rename = "range_from")]
+        RangeFrom { start: u16 },
+    }
+
+    impl From<&BinRangeuint16> for BinRangeuint16 {
+        fn from(value: &BinRangeuint16) -> Self {
+            value.clone()
+        }
+    }
+
+    /// A type storing a range over `T`.
+    ///
+    /// This type supports ranges similar to the `RangeTo`, `Range` and
+    /// `RangeFrom` types in the standard library. Those cover `(..end)`,
+    /// `(start..end)`, and `(start..)` respectively.
+    #[derive(Clone, Debug, Deserialize, Serialize, schemars :: JsonSchema)]
+    #[serde(tag = "type")]
+    pub enum BinRangeuint32 {
+        /// A range unbounded below and exclusively above, `..end`.
+        #[serde(rename = "range_to")]
+        RangeTo { end: u32 },
+        /// A range bounded inclusively below and exclusively above,
+        /// `start..end`.
+        #[serde(rename = "range")]
+        Range { end: u32, start: u32 },
+        /// A range bounded inclusively below and unbounded above, `start..`.
+        #[serde(rename = "range_from")]
+        RangeFrom { start: u32 },
+    }
+
+    impl From<&BinRangeuint32> for BinRangeuint32 {
+        fn from(value: &BinRangeuint32) -> Self {
+            value.clone()
+        }
+    }
+
+    /// A type storing a range over `T`.
```

Comment:
@bnaecker FYI we need to think about how we document these things since they appear in the public docs...